src/tpm2/crypto/openssl/CryptRsa.c

```diff
@@ -1357,8 +1357,7 @@ CryptRsaEncrypt(
                     ERROR_RETURN(TPM_RC_FAILURE);
                 memcpy(tmp, label->buffer, label->size);
             }
-            // label->size == 0 is supported
-            if (EVP_PKEY_CTX_set0_rsa_oaep_label(ctx, tmp, label->size) <= 0)
+            if (label->size > 0 && EVP_PKEY_CTX_set0_rsa_oaep_label(ctx, tmp, label->size) <= 0)
```

Comment:
Can you move the EVP_PKEY_CTX_set0_rsa_oaep_label() into the block above it that already checks for label->size > 0.